lib/TPP/ConvertLinalgToTpp.cpp

```diff
@@ -35,31 +35,55 @@ struct ConvertGenericOpToTpp : public OpRewritePattern<linalg::GenericOp> {
   LogicalResult rewriteToTppOp(linalg::GenericOp linalgOp,
                                PatternRewriter &rewriter) const {
     SmallVector<Value> operands;
+    if (tpp::utils::isTppZero(linalgOp, &operands)) {
+      assert((operands.size() != 0 && operands.size() <= 2) &&
+             "zero expects one or two operands");
+
+      // Only take the output, the other operand might be a constant.
+      rewriter.replaceOpWithNewOp<tpp::ZeroOp>(linalgOp,
+                                               operands[operands.size() - 1],
+                                               operands[operands.size() - 1]);
+      return success();
+    }
+
+    // Clear operands as the previous checks might have captured some operations
+    // that are no longer relevant for the next check.
+    operands.clear();
```

Comment:
Do we have false positive? We should capture right before returning `true`. Can you share a test cases for this?

Reply:
There was but you're right that we shouldn't pollute `operands` on false match. Adapted the code to handle that at isTpp* level.